packages/plugin-electron-session/session.js

```diff
@@ -1,12 +1,39 @@
 const sessionDelegate = require('@bugsnag/plugin-browser-session')
+const Session = require('@bugsnag/core/session')
 
 const SESSION_TIMEOUT_MS = 60 * 1000
 
-module.exports = (app, BrowserWindow) => ({
+module.exports = (app, BrowserWindow, NativeClient) => ({
   load (client) {
+    // Ensure native session kept in sync with session changes
+    Session.prototype._track = new Proxy(Session.prototype._track, {
+      apply (target, _this, args) {
+        const result = target.apply(_this, args)
+        NativeClient.setSession(serializeForNativeEvent(_this))
+        return result
+      }
+    })
+
     // load the actual session delegate from plugin-browser-session
     sessionDelegate.load(client)
 
+    // Override the delegate to perform electron-specific synchronization
+    const defaultDelegate = client._sessionDelegate
+    client._sessionDelegate = {
+      startSession: (client, session) => {
+        defaultDelegate.startSession(client, session)
+        NativeClient.setSession(serializeForNativeEvent(client._session))
+      },
+      resumeSession: (client) => {
+        defaultDelegate.resumeSession(client)
+        NativeClient.setSession(serializeForNativeEvent(client._session))
+      },
+      pauseSession: (client) => {
+        defaultDelegate.pauseSession(client)
+        NativeClient.setSession(serializeForNativeEvent(client._session))
+      }
```

Comment:
These should return whatever the `defaultDelegate` methods return:

```suggestion
      startSession: (client, session) => {
        const result = defaultDelegate.startSession(client, session)
        NativeClient.setSession(serializeForNativeEvent(client._session))
        return result
      },
      resumeSession: (client) => {
        const result = defaultDelegate.resumeSession(client)
        NativeClient.setSession(serializeForNativeEvent(client._session))
        return result
      },
      pauseSession: (client) => {
        const result = defaultDelegate.pauseSession(client)
        NativeClient.setSession(serializeForNativeEvent(client._session))
        return result
      }
```